File: Ridy-taxi-app/android/rider/src/main/java/com/innomalist/taxi/rider/models/RequestDTO.kt
package com.innomalist.taxi.rider.models

import com.google.android.gms.maps.model.LatLng
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RequestDTO(
        var locations: Array<LocationWithName>,
        var services: Array<OrderedService>,
        var intervalMinutes: Int = 0
)

@JsonClass(generateAdapter = true)
data class OrderedService(
        var serviceId: Long,
        var quantity: Int
)

@JsonClass(generateAdapter = true)
data class LocationWithName(
        var loc: LatLng,
        var add: String
)

File: Ridy-taxi-app/ios/rider/rider/Models/DTO/RequestDTO.swift
//
//  RequestDTO.swift
//  rider
//
//  Created by Manly Man on 11/15/19.
//  Copyright © 2019 minimal. All rights reserved.
//

import Foundation
import MapKit

struct RequestDTO: Codable {
    var locations: LocationWithNames
    var services: OrderedServices
    var intervalMinutes: Int = 0
}

struct OrderedService: Codable {
    var serviceId: Int
    var quantity: Int
}

struct LocationWithName: Codable {
    public var loc: CLLocationCoordinate2D
    public var add: String
}

typealias LocationWithNames = [LocationWithName]
typealias OrderedServices = [OrderedService]
